### DISCUSSION TITLE
Custom location cannot be retrieved properly on Apple M1 MacBook

### DISCUSSION THREAD

**Discussion by xb2o** [Stellarium/stellarium#2155] [Q&A]
I have two Apple systems running both MacOS Monterey 12.1 and Stellarium 0.21.3:
1. iMac Retina 5K 27-inch Late 2015 with Intel i7 processor
2. MacBook Pro 16-inch 2021 with M1 Max processor

When I create a new location and add it to the list on the iMac, I can switch to another location in the list, then search for the new location I just saved and select it. All the coordinates and elevation are correct as they were saved.

When I do the same on the M1 MacBook, I can add the new location to the list, but when I switch to another location and then back to the new site, only the elevation is correctly retrieved; longitude and latitude are not the one I saved, but the ones of the initial location.

Are there some files I could check to see if the new location parameters are properly saved?

Here are the 3 steps I went through:

<img width="742" alt="Capture d’écran 2022-01-04 à 12 11 43" src="https://user-images.githubusercontent.com/22239111/148054459-4787a099-961c-45d0-9ab2-bb11c094bacd.png">
<img width="742" alt="Capture d’écran 2022-01-04 à 12 13 55" src="https://user-images.githubusercontent.com/22239111/148054471-cb3b0b70-9d4d-498a-b0e2-41e0610246cc.png">
<img width="742" alt="Capture d’écran 2022-01-04 à 12 39 38" src="https://user-images.githubusercontent.com/22239111/148054487-62620514-7dcd-4c4e-953a-b598b8f8ba31.png">

I would expect to see MySite's coordinates to be at N 45°9'46" / E 5°35'44" as originally saved. 
.

**Comment by alex-w** [comment]
I created new location “test” (N50, E90) on my MacBook Pro M1 (macOS 12.1), added the location into list, mark option “Use current location as default” and restart the planetarium. No problem. OK, switching to my default location, restart the planetarium. Checked the coordinates of “test” location - all OK.

**Reply by xb2o** [reply]
I have unchecked the "Use current location as default" because the new location I created is not my current location. I am planning for a trip and wanted to check the sky from that location.

**Reply by xb2o** [reply]
Thanks for testing on your side.

**Comment by gzotti** [comment]
It seems that the coordinate are visible in the setting elements, but the arrow on the world map is not updated, therefore the coordinates are not really accepted. How exactly did you enter the coordinates? Mouseclick and wheel, or textual? Dis you press enter to confirm then?

**Reply by xb2o** [reply]
I entered each coordinate by editing the text strings and press Enter, then I clicked on Add to list

**Reply by xb2o** [reply]
I also tried to adjust coordinates using the up/down arrows. But to round the seconds I had to patch the decimals with 00 using the keyboard. 
It works better !
I now get the coordinates almost OK, except the decimals in the seconds which are stored with their original values, not the ones I set : e.g. 46.71" instead of 46.00".
How to get rid of these decimals?

**Reply by xb2o** [reply]
I finally solved my problem by converting the coordinates to decimals values and entering directly the decimal values instead of editing/patching the °'" values and it was stored and retrieved properly.
Longitude : E 5° 35’ 44” = 5.595556
Latitude : N 45° 9’ 46” = 45.162778
It finally does not seem to be different between i7 and M1 based Macs.

So the remaining issue I bypassed is how to round the seconds value.
